src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -197,10 +213,13 @@ moduleToJs (Module _ coms mn _ imps exps reExps foreigns decls) foreign_ =
   -- | Generate code in the simplified JavaScript intermediate representation for an accessor based on
   -- a PureScript identifier. If the name is not valid in JavaScript (symbol based, reserved name) an
   -- indexer is returned.
-  accessor :: Ident -> AST -> AST
-  accessor (Ident prop) = accessorString $ mkString prop
-  accessor (GenIdent _ _) = internalError "GenIdent in accessor"
-  accessor UnusedIdent = internalError "UnusedIdent in accessor"
+  moduleAccessor :: Ident -> AST -> AST
+  moduleAccessor (Ident prop) = moduleAccessorString prop
+  moduleAccessor (GenIdent _ _) = internalError "GenIdent in moduleAccessor"
+  moduleAccessor UnusedIdent = internalError "UnusedIdent in moduleAccessor"
+
+  moduleAccessorString :: Text -> AST -> AST
+  moduleAccessorString = accessorString . mkString . T.replace "'" "$prime"
```

Comment:
I think @natefaubion is more qualified to answer theses questions.

Reply:
I don't think I wrote this particular bit, but I think this is here because it's a tweak/extension of code that was already here which worked for CJS but needed to change for ESM. I don't see why we couldn't try to put it with other escaping code, I just don't know what the exact context is.

Reply:
It seems to me that this is necessary only because we switched from emitting `exports = { "foo'": foo$prime }` to `export { foo$prime }` and thus need to reference it with `Module_Name["foo$prime"]`, instead of `Module_Name["foo'"]` as we do currently. If we instead emit `export { foo$prime as "foo'" }` then this wouldn't have needed to change, I think. So it's either add mangling code here or modify the existing mangling code in `JS.Printer`, of which I think the latter is preferable. Decent chance I've gotten a detail wrong here though.

Reply:
> export { foo$prime as "foo'" }

I wasn't aware this was valid syntax. Wild!

Reply:
🤦 That's because it isn't, at least per the spec. It works in Node and I didn't bother to check beyond that. Mea culpa.

This is all fine then.

Reply:
It is according to the [most recent spec](https://tc39.es/ecma262/#prod-ModuleExportName) at least. But I don't think we can rely on the most recent spec.